resin/models/device.py

```diff
@@ -464,5 +464,110 @@ def register(self, app_name, uuid):
 
         return self.base_request.request(
             'device', 'POST', data=data,
+            endpoint=self.settings.get('pine_endpoint'), login=True
+        )
+
+    def restart(self, uuid):
+        """
+        Restart a device. This function only works if you log in using credentials or Auth Token.
+
+        Args:
+            uuid (str): device uuid.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+
+        """
+
+        device = self.get(uuid)
+        return self.base_request.request(
+            'device/{0}/restart'.format(device['id']),
+            'POST', endpoint=self.settings.get('api_endpoint'), login=True
+        )
+
+    def has_device_url(self, uuid):
+        """
+        Check if a device is web accessible with device urls
+
+        Args:
+            uuid (str): device uuid.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+
+        """
+
+        return self.get(uuid)['is_web_accessible']
+
+    def get_device_url(self, uuid):
+        """
+        Get a device url for a device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+
+        """
+        if not self.has_device_url:
+            raise exceptions.DeviceNoWebAccessible(uuid)
+
+        device_url_base = self.config.get_all()['deviceUrlsBase']
+        return 'https://{uuid}.{base_url}'.format(uuid=uuid, base_url=device_url_base)
+
+    def enable_device_url(self, uuid):
+        """
+        Enable device url for a device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+
+        """
+
+        if not self.has(uuid):
```

Comment:
I don't particularly like that this method is called `has`, it'd prefer `exists` or something. But anyway, not important.